gossipd/routing.c

```diff
@@ -1356,23 +1385,38 @@ bool routing_add_channel_update(struct routing_state *rstate,
 			hc->bcast.index
 				= gossip_store_add_private_update(rstate->gs,
 								  update);
-		} else
+			/* no need to separately track spam for private
+			 * channels */
+			hc->rgraph.index = hc->bcast.index;
+		} else {
 			hc->bcast.index = index;
+			hc->rgraph.index = index;
+		}
```

Comment:
These two branches are effectively the same?  `rgraph.index = bcast.index`?

Reply:
Added comments to indicate this behavior for non-spam messages.